src/utils/IBCStore.sol

```diff
@@ -35,26 +35,26 @@ contract IBCStore is IIBCStore, IICS24HostErrors, Ownable {
     /// @inheritdoc IIBCStore
     function commitPacket(IICS26RouterMsgs.Packet memory packet) public onlyOwner {
         bytes32 path = ICS24Host.packetCommitmentKeyCalldata(packet.sourceChannel, packet.sequence);
-        if (commitments[path] != 0) {
-            revert IBCPacketCommitmentAlreadyExists(
+        require(
+            commitments[path] == 0,
+            IBCPacketCommitmentAlreadyExists(
                 ICS24Host.packetCommitmentPathCalldata(packet.sourceChannel, packet.sequence)
-            );
-        }
+            )
+        );
 
         bytes32 commitment = ICS24Host.packetCommitmentBytes32(packet);
-        emit PacketCommitted(path, commitment);
         commitments[path] = commitment;
+        emit PacketCommitted(path, commitment);
```

Comment:
makes more logical sense for this to be at the end. This event should probably be removed on another issue to clean up unnecessary events

Reply:
Yes, I added an issue for implementing spec events, and specified in that issue that we should clean up events all over the place at the same time: https://github.com/cosmos/solidity-ibc-eureka/issues/108